osfoffline/filesystem_manager/sync_local_filesystem_and_db.py

```diff
@@ -167,6 +136,12 @@ def _determine_event_type(self, local, db):
             raise TypeError
 
         event = None
+
+        if local:
+            # TODO: find better way of ignoring files we want to ignore
+            if local.name.startswith('.') or local.name.endswith('~'):
```

Comment:
In the polling class, we silently exclude the `.Ds_Store` file from uploads (though it still generates a user message). We should start abstracting all "bad filename" checks to a single centralized utility function.

https://github.com/abought/OSF-Offline/blob/develop/osfoffline/polling_osf_manager/polling.py#L171-171